parts/k8s/kubernetesbase.t

```diff
@@ -50,25 +54,29 @@
         ]
       },
     {{end}}
-    {{ range $index, $element := .AgentPoolProfiles}}
-      {{if $index}}, {{end}}
-      {{if .IsWindows}}
-        {{if .IsVirtualMachineScaleSets}}
-          {{template "k8s/kuberneteswinagentresourcesvmss.t" .}}
-        {{else}}
-          {{template "k8s/kuberneteswinagentresourcesvmas.t" .}}
-        {{end}}
-      {{else}}
-        {{if .IsVirtualMachineScaleSets}}
-          {{template "k8s/kubernetesagentresourcesvmss.t" .}}
+    
+    {{if .AgentPoolProfiles }}
+      {{ range $index, $element := .AgentPoolProfiles}}
+        {{if $index}}, {{end}}
+        {{if .IsWindows}}
+          {{if .IsVirtualMachineScaleSets}}
+            {{template "k8s/kuberneteswinagentresourcesvmss.t" .}}
+          {{else}}
+            {{template "k8s/kuberneteswinagentresourcesvmas.t" .}}
+          {{end}}
         {{else}}
-          {{template "k8s/kubernetesagentresourcesvmas.t" .}}
+          {{if .IsVirtualMachineScaleSets}}
+            {{template "k8s/kubernetesagentresourcesvmss.t" .}}
+          {{else}}
+            {{template "k8s/kubernetesagentresourcesvmas.t" .}}
+          {{end}}
         {{end}}
       {{end}}
+      ,
     {{end}}
     {{if IsHostedMaster}}
       {{if not IsCustomVNET}}
-        ,{
```

Comment:
Just to make this changeset as clean as possible, can we use the original `,{` usage, even if it's arguably less readable?

Reply:
Assuming that there isnt any agentpool profiles, then this statement would start with a "," which arm doesnt seem to like :( had tested with different combo and had to settle with this change.